### PR TITLE
Special support for HFPretrainedConfig class objects

### PR DESCRIPTION
**Pull request by anijain2305**
~~@jansel This is for the HF models __get_attribute__ method~~

image

~~I am trying to inline the function, but then it calls object.__get_attribute__. And I am not sure, how to handle it.~~

Update - I also added special handling for HFPretrainedConfig. I am fine with whatever way is better. 

Further update - After offline discussion, decided to go with HFPretrainedConfig.

### CODE REVIEW

File: torchdynamo/variables/misc.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def call_method(
         inner_fn = self.const_getattr(self, name)
         if inner_fn is object.__init__:
             return LambdaVariable(identity, **options)
+        elif inner_fn is object.__getattribute__:
+            unimplemented(f"Cant handle object getattribute")
```
Comment (anijain2305) [inline torchdynamo/variables/misc.py:67]: Not sure how to handle this.

Comment (voznesenskym) [reply torchdynamo/variables/misc.py:67]: Check out the discussion on https://github.com/pytorch/torchdynamo/pull/454/files?

Comment (anijain2305) [reply torchdynamo/variables/misc.py:67]: Added a special handling for PreTrainedConfig

File: torchdynamo/variables/dicts.py [hunk 2]
```diff
@@ -279,3 +279,32 @@ def var_getattr(self, tx, name: str) -> "VariableTracker":
                 assert variables.ConstantVariable.is_literal(defaults[name])
                 return variables.ConstantVariable(defaults[name]).add_options(self)
         super(DataClassVariable, self).var_getattr(tx, name)
+
+
+class HFPretrainedConfigVariable(VariableTracker):
```
Comment (anijain2305) [inline torchdynamo/variables/dicts.py:284]: This is an alternate way - special handling.

File: torchdynamo/variables/user_defined.py [hunk 3]
```diff
@@ -281,7 +281,24 @@ def var_getattr(self, tx, name):
         options = VariableTracker.propagate(self)
         value = self.value
         source = AttrSource(self.source, name) if self.source else None
-        self._check_for_getattribute()
+
+        try:
```
Comment (anijain2305) [inline torchdynamo/variables/user_defined.py:285]: This is a more general way - however with issues that I am not sure how to resolve.

File: torchdynamo/exc.py [hunk 4]
```diff
@@ -67,7 +67,7 @@ def add_to_stats(self, category="unimplemented"):
 
 
 def unimplemented(msg: str):
-    if config.debug:
+    if True or config.debug:
```
Comment (jansel) [inline torchdynamo/exc.py:70]: Revert this?

**Review by jansel — APPROVED** [review]
I think special casing pretrained config fine way to handle it.

Handling getattribute more generally would be possible, but I don't suspect it will be very common.

**Pull request merged**